walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -128,46 +150,137 @@ private boolean attendeesIntersect(
     return false;
   }
 
+  /**
+   * Inserts a time range into a SortedMap containing an ordered overlapping set of times.
+   *
+   * @param rawSet The set of times
+   * @param timeRange The range to insert
+   */
+  private void insertTimeRange(SortedMap<Integer, TimeRange> rawSet, TimeRange timeRange) {
+    if (rawSet.containsKey(timeRange.start())) {
+      // If an event with the same start time is in the set,
+      // keep the longest event
+      TimeRange setTime = rawSet.get(timeRange.start());
+      if (timeRange.duration() > setTime.duration()) {
+        rawSet.put(timeRange.start(), timeRange);
+      }
+    } else {
+      rawSet.put(timeRange.start(), timeRange);
+    }
+  }
+
   /**
    * Converts a conflict set which may contain overlapping intervals into one that contains none.
    * Maximum runtime O(E).
    *
    * @param rawConflictSet
    * @return an ordered conflict set with no overlapping invervals.
    */
-  private Collection<TimeRange> cleanConflictSet(Collection<TimeRange> rawConflictSet) {
+  private List<TimeRange> cleanConflictSet(Collection<TimeRange> rawConflictSet) {
     List<TimeRange> conflictSet = new ArrayList<TimeRange>();
-    // Temporary TimeRange construction information
-    boolean tempRangeFlag = false;
-    int startTime = 0, endTime = 0;
+    FlatRangeSetBuilder builder = new FlatRangeSetBuilder();
 
     for (TimeRange rawRange : rawConflictSet) {
-      if (!tempRangeFlag) {
+      builder.add(rawRange);
+    }
+
+    return builder.getFlatRangeSet();
+  }
+
+  /**
+   * Combines two existing sorted and cleaned conflict sets. This is basically merging with extra
+   * considerations for end times. Runtime O(E).
+   *
+   * @param setA the first conflict set
+   * @param setB the second conflict set
+   * @return the union of the two conflict sets
+   */
+  public List<TimeRange> combineConflictSets(List<TimeRange> setA, List<TimeRange> setB) {
+    FlatRangeSetBuilder builder = new FlatRangeSetBuilder();
+    int indexA = 0, indexB = 0;
+
+    // Merge and deal with any overlapping intervals.
+    while (indexA < setA.size() && indexB < setB.size()) {
+      TimeRange rawRange = null;
+      if (setA.get(indexA).start() <= setB.get(indexB).start()) {
+        rawRange = setA.get(indexA);
+        ++indexA;
+      } else {
+        rawRange = setB.get(indexB);
+        ++indexB;
+      }
+
+      builder.add(rawRange);
+    }
+
+    while (indexA < setA.size()) {
+      builder.add(setA.get(indexA));
+      ++indexA;
+    }
+
+    while (indexB < setB.size()) {
+      builder.add(setB.get(indexB));
+      ++indexB;
+    }
+
+    return builder.getFlatRangeSet();
+  }
+
+  /** Builds a flattened range set, where none of the elements overlap. */
+  private class FlatRangeSetBuilder {
+    private List<TimeRange> rangeSet;
+    // Temporary range
+    private int startTime;
+    private int endTime;
+    private boolean flag;
```

Comment:
I would recommend keeping the more descriptive `flag` name or leaving a comment with regards to its purpose.

Reply:
Will address with a comment.